governator-test-junit/src/main/java/com/netflix/governator/guice/test/AnnotationBasedTestInjectorManager.java

```diff
@@ -0,0 +1,223 @@
+import com.netflix.archaius.guice.Raw;
+import com.netflix.archaius.test.TestCompositeConfig;
+import com.netflix.archaius.test.TestPropertyOverride;
+import com.netflix.archaius.test.TestPropertyOverrideAnnotationReader;
+import com.netflix.governator.InjectorBuilder;
+import com.netflix.governator.LifecycleInjector;
+import com.netflix.governator.providers.SingletonProvider;
+
+public class AnnotationBasedTestInjectorManager {
+
+    private final LifecycleInjector injector;
+    private final List<Object> mocksToReset = new ArrayList<>();
+    private final List<Module> modulesForTestClass = new ArrayList<>();
+    private final List<Module> overrideModules = new ArrayList<>();
+    private final List<Key<?>> spyTargets = new ArrayList<>();
+    private final SettableConfig classLevelOverrides = new DefaultSettableConfig();
+    private final SettableConfig methodLevelOverrides = new DefaultSettableConfig();
+    private final TestPropertyOverrideAnnotationReader testPropertyOverrideAnnotationReader = new TestPropertyOverrideAnnotationReader();
+    private TestCompositeConfig testCompositeConfig;
+
+    public AnnotationBasedTestInjectorManager(Class<?> classUnderTest) {
+        inspectModulesForTestClass(classUnderTest);
+        inspectMocksForTestClass(classUnderTest);
+        inspectSpiesForTargetKeys(Elements.getElements(modulesForTestClass));
+        testCompositeConfig = new TestCompositeConfig(classLevelOverrides, methodLevelOverrides);
+        overrideModules.add(new ArchaiusTestConfigOverrideModule(testCompositeConfig));
+        injector = createInjector(modulesForTestClass, overrideModules);
+    }
+
+    /**
+     * Injects dependencies into the provided test object.
+     */
+    public void prepareTestFixture(Object testFixture) {
+        injector.injectMembers(testFixture);
+    }
+
+    public void cleanupMocks() {
+        for (Object mock : mocksToReset) {
+            Mockito.reset(mock);
+        }
+    }
+
+    public void cleanupInjector() {
+        injector.close();
+    }
+
+    protected LifecycleInjector createInjector(List<Module> modules, List<Module> overrides) {
+        return InjectorBuilder.fromModules(modules).overrideWith(overrides).createInjector();
+    }
+
+    private void inspectModulesForTestClass(Class<?> testClass) {
+        final List<Class<? extends Module>> moduleClasses = new ArrayList<>();
+
+        moduleClasses.addAll(getModulesForAnnotatedClass(testClass));
+        for (Class<?> parentClass : getAllSuperClassesInReverseOrder(testClass)) {
+            moduleClasses.addAll(getModulesForAnnotatedClass(parentClass));
+        }
+        for (Class<? extends Module> moduleClass : moduleClasses) {
+            try {
+                modulesForTestClass.add(moduleClass.newInstance());
+            } catch (InstantiationException | IllegalAccessException e) {
+                try {
+                    Constructor<?> constructor = moduleClass.getDeclaredConstructors()[0];
```

Comment:
Handling for non-public constructors. I'll iterate and search for the zero arg constructor as suggested.